contracts/helpers/BaseHarvester.sol

```diff
@@ -2,97 +2,200 @@
+
+    /**
+     * @notice Set the yieldBearingAsset data
+     * @param yieldBearingAsset address of the yieldBearingAsset
+     * @param asset address of the asset
+     * @param targetExposure target exposure to the yieldBearingAsset asset used
+     * @param minExposure minimum exposure within the Transmuter to the asset
+     * @param maxExposure maximum exposure within the Transmuter to the asset
+     * @param overrideExposures whether limit exposures should be overriden or read onchain through the Transmuter
+     */
+    function setYieldBearingAssetData(
+        address yieldBearingAsset,
         address asset,
         uint64 targetExposure,
-        uint64 overrideExposures,
-        uint64 maxExposureYieldAsset,
-        uint64 minExposureYieldAsset,
-        uint96 _maxSlippage
-    ) {
-        ITransmuter transmuter = IRebalancerFlashloan(_rebalancer).TRANSMUTER();
-        TRANSMUTER = transmuter;
-        rebalancer = IRebalancerFlashloan(_rebalancer);
-        accessControlManager = IAccessControlManager(transmuter.accessControlManager());
-        _setCollateralData(
-            collateral,
+        uint64 minExposure,
+        uint64 maxExposure,
+        uint64 overrideExposures
+    ) external onlyGuardian {
+        _setYieldBearingAssetData(
+            yieldBearingAsset,
             asset,
             targetExposure,
-            minExposureYieldAsset,
-            maxExposureYieldAsset,
+            minExposure,
+            maxExposure,
             overrideExposures
         );
-        _setMaxSlippage(_maxSlippage);
+    }
+
+    /**
+     * @notice Set the limit exposures to the asset linked to the yield bearing asset
+     * @param yieldBearingAsset address of the yield bearing asset
+     */
+    function updateLimitExposuresYieldAsset(address yieldBearingAsset) public virtual {
+        YieldBearingParams storage yieldBearingInfo = yieldBearingData[yieldBearingAsset];
+        if (yieldBearingInfo.overrideExposures == 2)
+            _updateLimitExposuresYieldAsset(yieldBearingInfo.asset, yieldBearingInfo);
+    }
+
+    /**
+     * @notice Set the max allowed slippage
+     * @param newMaxSlippage new max allowed slippage
+     */
+    function setMaxSlippage(uint96 newMaxSlippage) external onlyGuardian {
+        _setMaxSlippage(newMaxSlippage);
+    }
+
+    /**
+     * @notice Toggle the trusted status of an address
+     * @param trusted address to toggle the trusted status
+     */
+    function toggleTrusted(address trusted) external onlyGuardian {
+        isTrusted[trusted] = !isTrusted[trusted];
+    }
+
+    /**
+     * @notice Recover ERC20 tokens
+     * @param tokenAddress address of the token to recover
+     * @param amountToRecover amount to recover
+     * @param to address to send the recovered tokens
+     */
+    function recoverERC20(address tokenAddress, uint256 amountToRecover, address to) external onlyGuardian {
+        IERC20(tokenAddress).safeTransfer(to, amountToRecover);
```

Comment:
let's add an event

Reply:
Also add a test on this --> only guardian can claim and that the logic is working